libs-haskell/test-utils/DA/Test/Util.hs

```diff
@@ -44,7 +44,12 @@ withTempFileResource :: (IO FilePath -> TestTree) -> TestTree
 withTempFileResource f = withResource newTempFile snd (f . fmap fst)
 
 withTempDirResource :: (IO FilePath -> TestTree) -> TestTree
-withTempDirResource f = withResource newTempDir snd (f . fmap fst)
+withTempDirResource f = withResource newTempDir delete (f . fmap fst)
+  -- The delete action provided by `newTempDir` calls `removeDirectoryRecursively`
+  -- and silently swallows errors. SDK installations are marked read-only
+  -- which means that they don’t end up being removed which is obviously
+  -- not what we intend.
```

Comment:
Is there a particular reason why we install SDK marked read-only? I actually find that quite inconvenient when I need to garbage collect SDK installations every now and then. Shouldn't we rather change the installation process?

Reply:
Figuring out why something goes wrong for a user is already challenging enough. If we add to that people that accidentally modified their installation we end up in a very messy situation. Currently, it’s at least a very deliberate choice if you do that.

As for garbage collecting, I believe that is better solved by adding commands to the assistant for doing that and not by making it easier to delete things manually.

Reply:
That makes sense. Thanks.